### PR TITLE
Upgrade node to version 20

### PR DESCRIPTION
**Pull request by roshaanbajwa**
## Description

- Upgrade pipelines to use node version 20
- Update dependancies and relevant syntax
- Add missing reference images for backstop

## Checklist

- [x] Tested against our [testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [x] Follows our [coding standards and style guide](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/coding-standards.md)

### CODE REVIEW

**Comment by frankieroberto** [comment]
@roshaanbajwa did upgrading node require all these dependencies to be updated at the same time (which then affected the reference screenshots in the tests)? 😩

**Review by edwardhorsford — COMMENTED** [review]
We'll also want to update the version listed in `.nvmrc`

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -10,6 +10,11 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 20
+        registry-url: https://registry.npmjs.org/
```
Comment (frankieroberto) [inline .github/workflows/pull-request.yml:16]: Bump this to v4 of the action? Shouldn't need to specify the registry, as npmjs is the default?  Setting `cache: 'npm'` should also speed up the `npm install` step by using a cache between builds.

```suggestion
    - uses: actions/setup-node@v4
      with:
        node-version: 20
        cache: 'npm'
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
 
       - uses: actions/setup-node@v1
```
Comment (frankieroberto) [inline .github/workflows/release.yml:18]:
```suggestion
      - uses: actions/setup-node@v4
```

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (roshaanbajwa) [reply .github/workflows/pull-request.yml:16]: good idea, adding it in now

**Review by frankieroberto — APPROVED** [review]

**Review by the-nathan-smith — APPROVED** [review]

**Pull request merged**